### PR TITLE
Dependency changes for jdk11

### PR DESCRIPTION
**Pull request by daveheitz**
Force build with jdk11
Don't do reflection for VirtualMachine (in order to avoid illegal access errors) unless necessary
Added stack trace in jvm agent when it didn't start

Signed-off-by: David Heitzman <david@heitzman.net>

### CODE REVIEW

**Comment by rhuss** [comment]
Thanks a lot for the PR ! However as this is not a small one and I'm about to head off for vacations, it will take some time until I will be able to review it.

One question: You changed the version to `-jdk11` something, so is this PR not compatible to Java 8 anymore ?

**Comment by daveheitz** [comment]
I explicitly made it build with source and target of jdk 11 to ensure compatibility. I suppose that could be rolled back to make it work with both; if that's a requirement I can try it.
Take your time, though.

File: agent/jvm/src/main/java/org/jolokia/jvmagent/client/AgentLauncher.java [hunk 1]
```diff
@@ -60,12 +62,8 @@ public static void main(String... args) {
...
             try {
                 exitCode = dispatcher.dispatchCommand(vm,vmHandler);
-            } catch (InvocationTargetException e) {
-                throw new ProcessingException("InvocationTargetException",e,options);
-            } catch (NoSuchMethodException e) {
-                throw new ProcessingException("Internal: NoSuchMethod",e,options);
-            } catch (IllegalAccessException e) {
-                throw new ProcessingException("IllegalAccess",e,options);
+            } catch (AgentException e) {
+                throw new ProcessingException("AgentExceptin",e,options);
```
Comment (bfg111) [inline agent/jvm/src/main/java/org/jolokia/jvmagent/client/AgentLauncher.java]: I think AgentExceptin should be "AgentException"

**Comment by bfg111** [comment]
@daveheitz  @rhuss - Any chance that you have time to get this fixed/merged? Running into issues getting monitoring to work in JDK11 on a few projects and this would be *really* nice to have working.

**Comment by rhuss** [comment]
Sorry for the awful delay, I was not able to invest that much time in Jolokia that it would deserve. @grgrzybek has fixed the reflection issues in the meantime, too, and we are going to make a 1.7.0 release asap.

Sorry again and thanks for your contribution.

**Pull request closed without merging**